src/tigerbeetle.zig

```diff
@@ -286,31 +298,65 @@ pub const CreateTransferResult = enum(u32) {
     exceeds_credits = 54,
     exceeds_debits = 55,
 
-    // TODO(zig): This enum should be ordered by precedence, but it crashes
+    debit_account_already_closed = 65,
+    credit_account_already_closed = 66,
+
+    // Last item: id_already_failed = 68.
+
+    comptime {
+        const values = std.enums.values(CreateTransferResult);
+        const BitSet = std.StaticBitSet(values.len);
+        var set = BitSet.initEmpty();
+        for (0..values.len) |index| {
+            const result: CreateTransferResult = @enumFromInt(index);
+            stdx.maybe(result == values[index]);
+
+            assert(!set.isSet(index));
+            set.set(index);
+        }
+
+        // It's a non-ordered enum, we need to ensure
+        // there are no gaps in the numbering of the values.
+        assert(set.count() == set.capacity());
+    }
+
+    /// TODO(zig): This enum is be ordered by precedence, but it crashes
     // `EnumSet`, and `@setEvalBranchQuota()` isn't propagating correctly:
     // https://godbolt.org/z/6a45bx6xs
     // error: evaluation exceeded 1000 backwards branches
     // note: use @setEvalBranchQuota() to raise the branch limit from 1000.
     //
-    // Workaround:
-    // https://github.com/ziglang/zig/blob/66b71273a2555da23f6d706c22e3d85f43fe602b/lib/std/enums.zig#L1278-L1280
-    imported_event_expected = 56,
-    imported_event_not_expected = 57,
-    imported_event_timestamp_out_of_range = 58,
-    imported_event_timestamp_must_not_advance = 59,
-    imported_event_timestamp_must_not_regress = 60,
-    imported_event_timestamp_must_postdate_debit_account = 61,
-    imported_event_timestamp_must_postdate_credit_account = 62,
-    imported_event_timeout_must_be_zero = 63,
+    // As a workaround we generate a new Ordered enum to be used in this case.
+    pub const Ordered = type: {
+        const values = std.enums.values(CreateTransferResult);
+        var fields: [values.len]std.builtin.Type.EnumField = undefined;
+        for (0..values.len) |index| {
+            const result: CreateTransferResult = @enumFromInt(index);
+            fields[index] = .{
+                .name = @tagName(result),
+                .value = index,
+            };
+        }
```

Comment:
That's a nice idea.

> Since the precedence ordering is really just "documentation"

One side effect of this change that I liked was that the client bindings are now also generated according to precedence. It can be quite helpful when reading the code (in other languages) because it matches the docs.

Reply:
Good point; I completely forgot about how it would impact the client implementation!